app/src/main/java/com/example/lotto649/Views/Fragments/HomeFragment.java

```diff
@@ -85,7 +87,23 @@ public void onEventsWaitListChanged() {
             @Override
             public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
                 EventModel event = (EventModel) adapterView.getItemAtPosition(i);
-                eventsController.editEvent(event);
+
+                String eventId = event.getEventId();
+                String deviceId = Settings.Secure.getString(getContext().getContentResolver(), Settings.Secure.ANDROID_ID);
+                String organizerId = event.getOrganizerId();
+                Bundle bundle = new Bundle();
+                bundle.putString("firestoreEventId", eventId);
+
+                if (Objects.equals(organizerId, deviceId)) {
+                    OrganizerEventFragment frag = new OrganizerEventFragment();
+                    frag.setArguments(bundle);
+                    getActivity().getSupportFragmentManager().beginTransaction()
+                            .replace(R.id.flFragment, frag, null)
+                            .addToBackStack(null)
+                            .commit();
+                } else {
+                    eventsController.editEvent(event);
```

Comment:
how does the organizer edit thier event now?

Reply:
There is an edit event button in the Organizer Event Fragment which will lead them to the edit fragment